fundingmanager.go

```diff
@@ -920,38 +1025,45 @@ func (f *fundingManager) handleInitFundingMsg(msg *initFundingMsg) {
 
 	fndgLog.Infof("Initiating fundingRequest(localAmt=%v, remoteAmt=%v, "+
 		"capacity=%v, numConfs=%v, addr=%v, dustLimit=%v)", localAmt,
-		msg.pushAmt, capacity, numConfs, msg.peer.addr.Address,
-		ourDustLimit)
+		msg.pushAmt, capacity, numConfs, msg.peerAddress, ourDustLimit)
 
 	// Initialize a funding reservation with the local wallet. If the
 	// wallet doesn't have enough funds to commit to this channel, then
 	// the request will fail, and be aborted.
 	reservation, err := f.wallet.InitChannelReservation(capacity, localAmt,
-		nodeID, msg.peer.addr.Address, uint16(numConfs), 4,
-		ourDustLimit, msg.pushAmt)
+		peerKey, msg.peerAddress.Address, uint16(numConfs), 4, ourDustLimit, msg.pushAmt)
 	if err != nil {
 		msg.err <- err
 		return
 	}
 
 	// Obtain a new pending channel ID which is used to track this
 	// reservation throughout its lifetime.
-	msg.peer.pendingChannelMtx.Lock()
-	chanID := msg.peer.nextPendingChannelID
-	msg.peer.nextPendingChannelID++
-	msg.peer.pendingChannelMtx.Unlock()
+	peer := f.findPeer(peerKey)
+	if peer == nil {
+		nilPeerMessage := "Unable to find peer. Peer is nil."
+		fndgLog.Errorf(nilPeerMessage)
+		msg.err <- errors.Errorf(nilPeerMessage)
+		return
+	}
+
+	peer.pendingChannelMtx.Lock()
+	chanID := peer.nextPendingChannelID
```

Comment:
My comment in the other PR about moving this logic into the peer itself also applies here.